torch/distributed/distributed_c10d.py

```diff
@@ -74,8 +77,14 @@ def __new__(cls, name):
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
             raise ValueError("Invalid backend: '{}'".format(name))
+        elif value != Backend.GLOO and value != Backend.NCCL and value != Backend.MPI:
+            value, _ = getattr(Backend, name.upper())
         return value
 
+    @classmethod
+    def register_new_backend(cls, model_name, instantiating_interface_name):
```

Comment:
> model_name

maybe module_name?

Reply:
> @ftian1 , @pietern
> Сould you please clarify how new approach will look from user script point of view?
> My current understanding is that user need to add one additional import and correctly specify backend name for init_process_group.
> 
> import torch.nn.parallel
> import torch.distributed as dist
> import <transport>
> 
> ...
> backend = '<transport_name>'
> dist.init_process_group(backend, ...)
> model = torch.nn.parallel.DistributedDataParallel(model, ...)
> ...

yes, you are right. Just pls note the backend paramter passed in dist.init_process_group() should be same with the one registered through Backend.register_new_backend().

and module_name is my typo. I will fix it soon